### PR TITLE
The 515 MegaPR early downport

### PR DESCRIPTION
**Pull request by cadyn**
## About The Pull Request

Early downport of https://github.com/VOREStation/VOREStation/pull/15739
If these changes are desired before the upstream PR is merged, I recommend test merging this. Make sure you set the engine to BYOND 515.1620.

## Changelog

:cl:
refactor: 515 Compatability
refactor: Nuke observer datums in favor of comsigs
refactor: GC Update
refactor: Mob stale references dealt with and set to GC normally
refactor: bicon and /icon ditched in favor of icon2html
refactor: Asset delivery redone
/:cl:

### CODE REVIEW

**Comment by Kashargul** [comment]
Make sure to check for any chompedits using \icon in this case. also integration test will go horrible if that is 1:1 code from virgo as chomp edit files will need adjustments. There might be a few switch cases that need to be converted to if-else.

**Comment by Raeschen** [comment]
> Make sure to check for any chompedits using \icon in this case. also integration test will go horrible if that is 1:1 code from virgo as chomp edit files will need adjustments. There might be a few switch cases that need to be converted to if-else.

A cursory glance couldn't find much \icon in modular_chomp on my end

**Comment by Raeschen** [comment]
Local tests i did on this seem mostly okay

Putting some info here for future reminders:

>Caches MUST be cleared after this upgrade.
>Clients may still be able to use 514 (stable)
>Clients using BETA 515 MUST clear IE's cache in windows!
>> "Delete cookies or temporary files" in start menu
>> General > Browsing history > Delete... button.
>Clients may benefit from using Fix TGChat and clearing TGChat's cache

**Comment by CHOMPStation2** [comment]
#### Test Merge Deployed

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 8824d1a13e472a633b513bc83b81258b98fc2f29
Pull Request: 386fe7ca5c762370b9e82b1efd21c7d2e8a74b6c
Server: 565d19061de99e8a4ff729f3052560ac003c7969

##### Comment

</details>

File: _build_dependencies.sh [hunk 1]
```diff
@@ -6,16 +6,16 @@
 export RUST_G_VERSION=3.1.0
 
 # byond version
-export BYOND_MAJOR=514
-export BYOND_MINOR=1589
+export BYOND_MAJOR=515
+export BYOND_MINOR=1620
```
Comment (ItsSelis) [inline _build_dependencies.sh:10]:
```suggestion
export BYOND_MINOR=1630
```

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 3f80966a8e75f017cfd37be67adf79f3afd93798
Pull Request: 386fe7ca5c762370b9e82b1efd21c7d2e8a74b6c
Server: 663c37b33e6a6c42f147336c608082f2d4f27f22

##### Comment

</details>

**Comment by ItsSelis** [comment]
this runtime appears very often:
Runtime in code/_helpers/unsorted.dm,1536: observer_move overridden. Use override = TRUE to suppress this warning.

File: code/_helpers/text.dm [hunk 2]
```diff
@@ -355,7 +355,7 @@
 		return tagdesc
 	if(!text_tag_cache[tagname])
 		var/icon/tag = icon(text_tag_icons, tagname)
-		text_tag_cache[tagname] = bicon(tag, TRUE, "text_tag")
+		text_tag_cache[tagname] = icon2html(tag, C, extra_classes = "text_tag")
```
Comment (Raeschen) [inline code/_helpers/text.dm:358]: OOC, DCHAT, LOOC.... icons aren't visible ingame, i suspect it has something to do with this.

**Comment by ItsSelis** [comment]
another runtime that appeared with observers:
Runtime in code/datums/observation/unequipped.dm,39: Cannot read null.comp_lookup

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 3f80966a8e75f017cfd37be67adf79f3afd93798
Pull Request: f89c347d05a4026d78a6e9e38dc6fc9703f27d9a
Server: eb1fdf5839d32c00d102b2e3392b7a6795de7293

##### Comment

</details>

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -736,7 +736,7 @@ modules/mob/living/carbon/human/life.dm if you die, you will be zoomed out.
 			H.toggle_zoom_hud()	// If the user has already limited their HUD this avoids them having a HUD when they zoom in
 		H.set_viewsize(viewsize)
 		zoom = 1
-		GLOB.moved_event.register(H, src, PROC_REF(zoom))
+		RegisterSignal(H, COMSIG_OBSERVER_MOVED, PROC_REF(zoom))
```
Comment (Raeschen) [inline code/game/objects/items.dm:739]: When this calls back zoom, zoom's src seems to be null.

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: ddc4f199c80a06105000196f1e2460f34cc38d86
Pull Request: f89c347d05a4026d78a6e9e38dc6fc9703f27d9a
Server: 48597a944b94689e7855b5f4e2ef21ab9e277bed

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: aea2d14a86e14ec25ccf1924973d43404ecb84a9
Pull Request: f89c347d05a4026d78a6e9e38dc6fc9703f27d9a
Server: 2611c4117c41eab056610aad861e9e1184bb4f5d

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 19b8d6e4f29afaa72ea246795fccc478874998b6
Pull Request: f89c347d05a4026d78a6e9e38dc6fc9703f27d9a
Server: 42ae7f02a855991df497a75986e1394534d4dbf0

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 19b8d6e4f29afaa72ea246795fccc478874998b6
Pull Request: 3c7a593490bd3162d0d8288e0a8406b644382bcd
Server: d0adcdcc351df36e86318bd011f3ff427d47799e

##### Comment

</details>

**Comment by ItsSelis** [comment]
.tgs.yml has to be bumped up accordingly too :x

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 7c388db42bf7509578c7a1cc103d490b18483365
Pull Request: 3c7a593490bd3162d0d8288e0a8406b644382bcd
Server: 8e18a9601a1e60442203e59e6eb71a7a232c9416

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: d54171efc8a702aa846a2e01e1c83782497daf4f
Pull Request: 3c7a593490bd3162d0d8288e0a8406b644382bcd
Server: 6a10d3fc00cec7b762a5b1c64010f8bb955380d2

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: dd3355c8c87bf182bce08d422f78fa4a818d1555
Pull Request: 3c7a593490bd3162d0d8288e0a8406b644382bcd
Server: e10584616a939bcefa77c00c47a66ca7413745bd

##### Comment

</details>

File: code/modules/overmap/ships/computers/ship.dm [hunk 4]
```diff
@@ -77,7 +77,8 @@ somewhere on that shuttle. Subtypes of these can be then used to perform ship ov
 		L.looking_elsewhere = 1
 		L.handle_vision()
 	user.set_viewsize(world.view + extra_view)
-	GLOB.moved_event.register(user, src, /obj/machinery/computer/ship/proc/unlook)
+	RegisterSignal(user, COMSIG_OBSERVER_MOVED, /obj/machinery/computer/ship/proc/unlook)
+	RegisterSignal(user, COMSIG_OBSERVER_MOVED, /obj/machinery/computer/ship/proc/unlook, override = TRUE) //This needs override or else it will spam runtimes because it is called repeatedly.
```
Comment (Raeschen) [inline code/modules/overmap/ships/computers/ship.dm:81]: Repeated intentionally?

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: dd3355c8c87bf182bce08d422f78fa4a818d1555
Pull Request: 11fec17e9a188268d5f8aeedb7126785dafb3196
Server: 96f4e023c1b82795fb129efcb41e6b3cfbc7d3c5

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: dd3355c8c87bf182bce08d422f78fa4a818d1555
Pull Request: 8931f510f0735b245e7607f6a8ac0415a3b89d01
Server: 8dfe7b4453bccd395e196ba067cb3cdb7de65452

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 04cbe1be8b706661b8eb78cf86beddeeb5f9fb57
Pull Request: 8931f510f0735b245e7607f6a8ac0415a3b89d01
Server: 4d19bc3053eb124b445ece25f71a2ca94b2a272f

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 0bdea00d79f7c93c8267c7a67d361d9159e8dccb
Pull Request: 09e4c29d8101ecf63f22564c95c37d58e95d4e8f
Server: ef366795a464dc6218c9155c26b5c0953220d576

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 88e2d121d4eaf23ad57a001e2afe3a8cee51b07e
Pull Request: 09e4c29d8101ecf63f22564c95c37d58e95d4e8f
Server: c68d0a2240657b473f82214a33c22a00d9289114

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 2d80c806bf141c1700eb42d5c4ceb76f4dbb93c7
Pull Request: 09e4c29d8101ecf63f22564c95c37d58e95d4e8f
Server: 8f0e05b9ad2ecd86c45dfcc947bbdc5e68c8c8bc

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: aba18c8c07987fe0ef6f7059d9c3a78ea8b07185
Pull Request: 09e4c29d8101ecf63f22564c95c37d58e95d4e8f
Server: 82aadfd906e2cbdc4ff0f9ad1f2dc07d0f6c7c76

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: f3987c9bfea916255d32edb59971209d4edac3d9
Pull Request: 09e4c29d8101ecf63f22564c95c37d58e95d4e8f
Server: 67161a9dc710db71b2cc56fc0ec8084e8ee629ef

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 12370f48558661c6c4d3599f6f5919445ab762bd
Pull Request: 09e4c29d8101ecf63f22564c95c37d58e95d4e8f
Server: c137ca3cc180f392a286b70f7c5ca40e88b20ade

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 7eabb7b371100a63c4e0197e6490c767d0151d7f
Pull Request: 09e4c29d8101ecf63f22564c95c37d58e95d4e8f
Server: da1b9e7250ce0fe78508c59ca51a59df439edea4

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 50405e149fa937d0047bd03b1afe764acaa88276
Pull Request: 09e4c29d8101ecf63f22564c95c37d58e95d4e8f
Server: 3145f61585140edecc4b45ce33144d3012978e4b

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 1e4c8a0ca5f9b6be1c14d54ad3720d7a091b9c45
Pull Request: 09e4c29d8101ecf63f22564c95c37d58e95d4e8f
Server: bcfdd9951180371896c8464e7f6963fc8d602370

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: ed427a914da9c2d42ca75e50f250f8baf51de579
Pull Request: 09e4c29d8101ecf63f22564c95c37d58e95d4e8f
Server: 3006bc894c6f0aeadae3db5b9ad986046c59f15b

##### Comment

</details>

**Comment by Raeschen** [comment]
Useful with this PR is, in displayed order;
#7791 
#7795 
#7799 
#7810

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: ff5efcf436e2d9ebe2a796d1ccd42996ebed823d
Pull Request: 09e4c29d8101ecf63f22564c95c37d58e95d4e8f
Server: 3d7365efd50ee23f826c2a86c9a58858674826e1

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: ff5efcf436e2d9ebe2a796d1ccd42996ebed823d
Pull Request: 7498e24b80e1fcb974d1a7694cc9c669dd33a804
Server: bcf078082bb2f06a33ee6492aff10c47dab734db

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 65c2e364ceb29348570a599c07ce9df3174653df
Pull Request: fb7aaa73d8943f72de20d7f4c033f3e637100f65
Server: 1c8ad828bdbea743beff846c730fee5ddd1a79a8

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 2a70e36ff81d8d64b1e66ec6d526cb548489e6fe
Pull Request: fb7aaa73d8943f72de20d7f4c033f3e637100f65
Server: 193d1150a3f11967ee9872ee559b3ba63d809a95

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: d3457cffe4ed40b960ba77db3dfe748382d3a880
Pull Request: fb7aaa73d8943f72de20d7f4c033f3e637100f65
Server: 798b3a3d63c77ec415e94c7367c0f837f670ef1c

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 8869abde135c07095006c499c949b326be9c574c
Pull Request: fb7aaa73d8943f72de20d7f4c033f3e637100f65
Server: 3dc3fecec8256e878e927eaf33ee891767e7c594

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 8869abde135c07095006c499c949b326be9c574c
Pull Request: 835d4811c781700da5c9abd18e94323667d21827
Server: 0a51071fa897337e02e9a9ab00143eeb2988211d

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 65113c9809788f5aae17b1dbfeb34dc98bc8f53a
Pull Request: 835d4811c781700da5c9abd18e94323667d21827
Server: 0bebd86bacdb8b838b4acbead5b9d29445548997

##### Comment

</details>

**Comment by CHOMPStation2** [comment]
#### Test Merge Updated

<details><summary>Details</summary>

##### Server Instance
CHOMP2

##### Revision
Origin: 65113c9809788f5aae17b1dbfeb34dc98bc8f53a
Pull Request: cda5f1f0169f362ef8529cc80b9102fd14f56158
Server: c6c3487e575b77ac1c0b6de6abb60647bd43ec94

##### Comment

</details>

**Comment by Raeschen** [comment]
> I'm feeling pretty good about where this is now. I would say, give it ~48 hours after test merge is updated to latest commit to ensure stability on chomp and then merge it. The PR on virgo is already in sync with where this is now.

Thankyou so much. As well as putting up with where i've messed up a little in places. I learnt a lot trying to assist with this.

**Pull request merged**